### PR TITLE
Add support for new properties of azurerm_cosmosdb_account

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new properties `analyticalStorageConfiguration.schemaType`, `capacity.totalThroughputLimit`, `defaultIdentity` and `backupPolicy.storageRedundancy` for Cosmos DB Account.

--- PASS: TestAccCosmosDBAccount_updateCapacity (2046.13s)
--- PASS: TestAccCosmosDBAccount_updateAnalyticalStorage (2023.45s)
--- PASS: TestAccCosmosDBAccount_updateDefaultIdentity (2054.76s)
--- PASS: TestAccCosmosDBAccount_backup (2038.46s)

image
image

### CODE REVIEW

File: internal/services/cosmos/cosmosdb_account_resource.go [hunk 1]
```diff
@@ -110,6 +110,53 @@ func resourceCosmosDbAccount() *pluginsdk.Resource {
 				}, true),
 			},
 
+			"analytical_storage_configuration": {
```
Comment (katbyte) [inline internal/services/cosmos/cosmosdb_account_resource.go:113]: we can probably make this
```suggestion
			"analytical_storage": {
```

File: internal/services/cosmos/cosmosdb_account_resource.go [hunk 2]
```diff
@@ -110,6 +110,53 @@ func resourceCosmosDbAccount() *pluginsdk.Resource {
...
+				},
+			},
+
+			"capacity": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"total_throughput_limit": {
```
Comment (katbyte) [inline internal/services/cosmos/cosmosdb_account_resource.go:138]: what are the units here?

File: internal/services/cosmos/cosmosdb_account_resource.go [hunk 3]
```diff
@@ -110,6 +110,53 @@ func resourceCosmosDbAccount() *pluginsdk.Resource {
...
+							Type:         pluginsdk.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntAtLeast(-1),
+						},
+					},
+				},
+			},
+
+			"default_identity": {
```
Comment (katbyte) [inline internal/services/cosmos/cosmosdb_account_resource.go:147]: should this be
```suggestion
			"default_identity_type": {
```

File: website/docs/r/cosmosdb_account.html.markdown [hunk 4]
```diff
@@ -164,6 +170,19 @@ The following arguments are supported:
 
 ---
 
+A `analytical_storage_configuration` block supports the following:
+
+* `schema_type` - (Required) The schema type of the Analytical Storage for this Cosmos DB account. Possible values are `FullFidelity` and `WellDefined`.
+
+---
+
```
Comment (katbyte) [inline website/docs/r/cosmosdb_account.html.markdown:178]:
```suggestion
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei - comments comments left inline

File: internal/services/cosmos/cosmosdb_account_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_account_resource.go:113]: Updated

File: internal/services/cosmos/cosmosdb_account_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_account_resource.go:138]: The unit is RU/s. I added the description of unit in md file.

File: internal/services/cosmos/cosmosdb_account_resource.go (hunk 3 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_account_resource.go:147]: Updated

File: website/docs/r/cosmosdb_account.html.markdown (hunk 4 above)
Comment (neil-yechenwei) [reply website/docs/r/cosmosdb_account.html.markdown:178]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please take another look. Thanks in advance.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 🏗️

**Pull request merged**